src/merkle.rs

```diff
@@ -447,7 +447,7 @@ impl<
         );
 
         let store = S::new_from_slice(tree_len, &data).context("failed to create data store")?;
-        let root = store.read_at(data.len() - 1)?;
+        let root = store.read_at(store.len() - 1)?;
```

Comment:
Are you sure these are compatible?  I thought one was in element counts and one was in data/byte counts(?)

Reply:
Thanks, for pointing on this. This is a place, where I'm not sure completely, TBH. But two following thoughts caused me to introduce this change:
- I have double-checked that `from_tree_slice` constructor doesn't have tests neither in this repository, nor in `rust_fil_proofs` - so it is impossible to check compatibility actually;
- `data` from this constructor is actually a byte slice that, as far as I understand this, represents serialised storage in some custom format. But `store` is an instance of storage, created after deserialising `data` via `S::new_from_slice`, and if we want to get last element from store (which is root of tree), we need to use indexing of store - not data. 

I added tests for this constructor, where base binary tree with vector storage is instantiated with 4 leaves (each leaf is a 16-bytes array) and basic functionality of the tree is evaluated. This tree has 7 elements in total (4 leaves + 2 nodes + 1 root), which gives 7 * 16 = 112 bytes slice for serialised tree (our `data`). With previous version of code I was getting panic:

```
thread 'test_merkle_constructors::test_from_tree_slice_group' panicked at 'index out of bounds: the len is 7 but the index is 111', src/store/vec.rs:86:12
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
test test_merkle_constructors::test_from_tree_slice_group ... FAILED
```
specifically at this place, when root was being read. And it is reasonable, since we use `data` which is 112 bytes slice, instead of `store` (of `VecStore` type) which is actually vector of elements (`Element` type, which in my case is `Item` - 16-bytes array), so according to the logic in mind, we need to take last `Item` from our `store`, which has index: `[store.len() - 1]`.